### PR TITLE
Migrated  `pkg/scheduler/framework/plugins/examples/` to use contextual logging

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

This PR will modify the following:
- Migrated `pkg/scheduler/framework/plugins/examples/stateful` to use contextual logging

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/91633#issuecomment-1467645065

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging
```

/sig instrumentation
/wg structured-logging
/area logging
/sig scheduling

~Dependent on PR https://github.com/kubernetes/kubernetes/pull/116842
Wait for the PR https://github.com/kubernetes/kubernetes/pull/116842 to merge first.~

### CODE REVIEW

File: pkg/scheduler/framework/plugins/examples/stateful/stateful.go [hunk 1]
```diff
@@ -82,9 +82,9 @@ func (mp *MultipointExample) PreBind(ctx context.Context, state *framework.Cycle
 }
 
 // New initializes a new plugin and returns it.
-func New(config *runtime.Unknown, _ framework.Handle) (framework.Plugin, error) {
+func New(ctx context.Context, config *runtime.Unknown, fh framework.Handle) (framework.Plugin, error) {
```
Comment (chendave) [inline pkg/scheduler/framework/plugins/examples/stateful/stateful.go:85]: `fh framework.Handle` why this change is needed?

**Review by kerthcet — COMMENTED** [review]
/remove-kind feature
/approve

Also, since it's an example, I think there's no user-facing change introduced. Please remove the release note.

**Comment by mengjiao-liu** [comment]
> Also, since it's an example, I think there's no user-facing change introduced. Please remove the release note.

Removed.

File: pkg/scheduler/framework/plugins/examples/stateful/stateful.go (hunk 1 above)
Comment (mengjiao-liu) [reply pkg/scheduler/framework/plugins/examples/stateful/stateful.go]: I forgot to restore the original content that I changed before😅. Updated.
